internal/api/v2/controllers_transactions_list.go

```diff
@@ -1,39 +1,40 @@
 package v2
 
 import (
-	"net/http"
-
 	"errors"
 	"github.com/formancehq/go-libs/v2/api"
 	"github.com/formancehq/go-libs/v2/bun/bunpaginate"
 	"github.com/formancehq/ledger/internal/api/common"
 	ledgercontroller "github.com/formancehq/ledger/internal/controller/ledger"
+	"net/http"
 )
 
-func listTransactions(w http.ResponseWriter, r *http.Request) {
-	l := common.LedgerFromContext(r.Context())
+func listTransactions(paginationConfig common.PaginationConfig) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		l := common.LedgerFromContext(r.Context())
 
-	paginationColumn := "id"
-	if r.URL.Query().Get("order") == "effective" {
-		paginationColumn = "timestamp"
-	}
-
-	rq, err := getColumnPaginatedQuery[any](r, paginationColumn, bunpaginate.OrderDesc)
-	if err != nil {
-		api.BadRequest(w, common.ErrValidation, err)
-		return
-	}
+		paginationColumn := "id"
+		if r.URL.Query().Get("order") == "effective" {
+			paginationColumn = "timestamp"
+		}
 
-	cursor, err := l.ListTransactions(r.Context(), *rq)
-	if err != nil {
-		switch {
-		case errors.Is(err, ledgercontroller.ErrInvalidQuery{}) || errors.Is(err, ledgercontroller.ErrMissingFeature{}):
+		rq, err := getColumnPaginatedQuery[any](r, paginationConfig, paginationColumn, bunpaginate.OrderDesc)
+		if err != nil {
 			api.BadRequest(w, common.ErrValidation, err)
-		default:
-			common.HandleCommonErrors(w, r, err)
+			return
 		}
-		return
-	}
 
-	api.RenderCursor(w, *cursor)
+		cursor, err := l.ListTransactions(r.Context(), *rq)
+		if err != nil {
+			switch {
+			case errors.Is(err, ledgercontroller.ErrInvalidQuery{}) || errors.Is(err, ledgercontroller.ErrMissingFeature{}):
```

Comment:
`ledgercontroller.ErrInvalidQuery{}` and `ledgercontroller.ErrMissingFeature{}` are starting to look like common errors since you are adding this handling in all the list endpoints. Perhaps this logic could be moved to `common.HandleCommonErrors` ?